src/components/common/Modal/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect, useRef } from 'react'
+import ReactDOM from 'react-dom'
+import { ReactComponent as CloseSvg } from '../../../img/close.svg'
+import styles from './Modal.module.scss'
+
+interface ModalProps {
+  title?: string
+  open: boolean
+  children: React.ReactNode
+  onClose: () => void
+}
+export default function Modal({ title, open, children, onClose }: ModalProps) {
+  const wrapperRef = useRef(null)
+  useEffect(() => {
+    function onWrapperClicked(event) {
+      if (wrapperRef.current && wrapperRef.current === event.target) {
+        onClose()
+      }
+    }
+
+    document.addEventListener('mousedown', onWrapperClicked)
+    return () => {
+      document.removeEventListener('mousedown', onWrapperClicked)
+    }
+  }, [onClose, wrapperRef])
+
+  useEffect(() => {
+    document.body.style.overflow = open ? 'hidden' : 'unset'
```

Comment:
- It's for scrolling prevention, right? Probably good example where a comment would be useful :) 
- Probably doesn't matter for us but here is a potential bug if original overflow style on the body wasn't `unset`

Reply:
Added a comment. I thought about that potential bug, but I couldn't think about any other way to handle this, open to suggestions.

Reply:
Discussed in DM it's fine for now, we don't plan to change body overflow